sparse_strips/vello_common/src/tile.rs

```diff
@@ -322,7 +322,214 @@ impl Tiles {
 #[cfg(test)]
 mod tests {
     use crate::flatten::{Line, Point};
-    use crate::tile::Tiles;
+    use crate::tile::{Tile, Tiles};
+
+    #[test]
+    fn cull_line_at_top() {
+        let line = Line {
+            p0: Point { x: 3.0, y: -5.0 },
+            p1: Point { x: 9.0, y: -1.0 },
+        };
+
+        let mut tiles = Tiles::new();
+        tiles.make_tiles(&[line], 100, 100);
+
+        assert!(tiles.is_empty());
+    }
+
+    #[test]
+    fn cull_line_at_right() {
+        let line = Line {
+            p0: Point { x: 101.0, y: 0.0 },
+            p1: Point { x: 103.0, y: 20.0 },
+        };
+
+        let mut tiles = Tiles::new();
+        tiles.make_tiles(&[line], 100, 100);
+
+        assert!(tiles.is_empty());
+    }
+
+    #[test]
+    fn cull_line_at_bottom() {
+        let line = Line {
+            p0: Point { x: 30.0, y: 101.0 },
+            p1: Point { x: 35.0, y: 105.0 },
+        };
+
+        let mut tiles = Tiles::new();
+        tiles.make_tiles(&[line], 100, 100);
+
+        assert!(tiles.is_empty());
+    }
+
+    #[test]
+    fn partially_cull_line_exceeding_viewport() {
+        let line = Line {
+            p0: Point { x: -2.0, y: -3.0 },
+            p1: Point { x: 2.0, y: 1.0 },
+        };
+
+        let mut tiles = Tiles::new();
+        tiles.make_tiles(&[line], 100, 100);
+
+        assert_eq!(tiles.tile_buf, vec![Tile::new(0, 0, 0, true)]);
```

Comment:
It may be slightly faster to compare to a slice, sparing the allocation:

```suggestion
        assert_eq!(&tiles.tile_buf, &[Tile::new(0, 0, 0, true)]);
```